core/core-impl/src/main/java/ai/timefold/solver/core/config/solver/EnvironmentMode.java

```diff
@@ -19,6 +19,26 @@
  */
 @XmlEnum
 public enum EnvironmentMode {
+    /**
+     * This mode turns on variable tracking and all assertions to fail-fast on a bug in a {@link Move} implementation, a
+     * constraint, the engine itself or something else at an appalling performance cost.
+     * <p>
+     * Because it tracks genuine and shadow variables, it is able to report precisely what variables caused the corruption and
+     * report any missed {@link ai.timefold.solver.core.api.domain.variable.VariableListener} events.
+     * <p>
+     * When undo score corruption is detected, it throws an
+     * {@link ai.timefold.solver.core.api.solver.exception.UndoScoreCorruptionException}, which can be used to inspect the
+     * working solution state before, after and after the undo of a {@link Move}.
+     * <p>
+     * This mode is reproducible (see {@link #REPRODUCIBLE} mode).
+     * <p>
+     * This mode is intrusive because it calls the {@link InnerScoreDirector#calculateScore()} more frequently than a non assert
+     * mode.
+     * <p>
+     * This mode is appallingly slow.
+     */
```

Comment:
Besides the general comment on line breaks and formatting, I think maybe we're a bit more evolved now to use words such as "horrible" and "appaling" in user-facing documentation. :-) 
In this case, I suggest "This mode is by far the slowest of all the modes."

Also, somewhere in the body of the text, I would reference that it does everything that `FULL_ASSERT` does, plus X and Y. If you can reference an information that is already somewhere else, I prefer you do so - that way, you still have only a single point of truth for that information.